### ISSUE TITLE
Perfect IV's :)

### ISSUE THREAD

**Issue by arkdelkaos** [SwipeX/PokeMate#2]
You know what could be AWESOME? If PokeMate could **calculate the IV of every pokemon caught, and if its perfect (or >85%) keep it** 👍 
Ever better if you could just set a white-list of "perfect" pokemons to keep (some are just sausage meat even when they are perfect)
https://www.reddit.com/r/TheSilphRoad/comments/4tupg5/eli5_what_is_iv_and_what_is_perfect/

BTW:

```
Transferring 1/2 EKANS lvl 49
```

Being a lvl3 account when the pokemon was caught...I don't think that Ekans could be so hi-level. Maybe something's wrong there ;)

**Comment by SwipeX** [comment]
Some sorting/keeping can definitely be done with IV.

> Transferring 1/2 EKANS lvl 49

This is coming straight from the API. I would tell you that it's just a chance that you caught a level 49, and does not seem that crazy.

**Comment by SwipeX** [comment]
I will keep this issue open until something has been added to deal with IV transferring.

**Comment by arkdelkaos** [comment]
Could you show me an example of the api response when you catch a pokemon? :)

**Comment by SwipeX** [comment]
https://github.com/SwipeX/PokeMate/blob/master/src/main/java/dekk/pw/pokemate/tasks/CatchPokemon.java#L38
Edit: it is definitely possible.

**Comment by arkdelkaos** [comment]
OK....what you identify as "level" is the CP! :) I was going nuts with that one xD
The level, ad well as ATK, DEF and STM, are "hidden" stats. There is an algorithm that getting the 3 params, and the using the level as a multiplier, the game calculate the CP of a Pokemon. Sure, when you use stardust you "level up" the CP, but because you increase the pokemon level...and then the CP using that algorithm :)
I don't know if you can access those IV's directly with the API (that could simplify everything a lot!), but having the CP, HP, Stardust cost, and the trainer level, you can calculate the IV's back :)
Being able to filter between the bad pokemons and Hi IV's ones....could be awesome.
I'm looking at improving it myself, but I don't know what options do I have: `list.get(i).getSOMETHING()`
Its there some kind of functions list for every pokemon item? :)

**Comment by pashorizer** [comment]
Yes you can access all those fields with

```
encounterResult.wildPokemon.pokemonData
.individualStamina
.individualAttack
.individualDefense

```

The formular for IV is : 
`s + a + d * 100 / 45`

**Comment by arkdelkaos** [comment]
OMG...you made me happy :)
I'm going to test it, and then commit some changes ;)

**Comment by SwipeX** [comment]
After much refactoring this is done. Will be in next release.
